src/util/inc/tutil.h

```diff
@@ -110,6 +110,23 @@ extern "C" {
 
 #define POW2(x) ((x) * (x))
 
+#define MEMNCPY(dst, src, n)       \
+  do {                             \
+    assert((n) <= sizeof(dst));    \
+    strncpy((dst), (src), n);      \
+  } while (0)
+
+#define STRNCPY(dst, src, n)       \
+  do {                             \
+    assert((n) <= sizeof(dst));    \
+    int nn = n;                    \
+    if (nn >= sizeof(dst)) {       \
+      nn = sizeof(dst) - 1;        \
+    }                              \
+    strncpy((dst), (src), nn);     \
+    (dst)[nn] = 0;                 \
```

Comment:
if `n` is less than `sizeof(dst)`, this line modifies the `n + 1` byte, which is not expected.

Reply:
please be kindly noted, a MEMNCPY macro is also added (just lines before STRNCPY definition) which does not write null-terminator, and you can use MEMNCPY to handle the case you mentioned.